ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m

```diff
@@ -51,58 +50,56 @@ + (ADRegistrationInformation *)getRegistrationInformation:(id<MSIDRequestContext
+    if (!identity || CFGetTypeID(identity) != SecIdentityGetTypeID())
     {
-        if (error)
+        MSID_LOG_VERBOSE(context, @"Failed to retrieve WPJ identity.");
+        if (identity)
         {
-            *error = adError;
+            CFRelease(identity);
+            identity = NULL;
         }
-        MSID_LOG_ERROR(context, @"Failed to retrieve WPJ certificate. Error code: %ld", (long)adError.code);
-        goto _error;
+        
+        return NULL;
     }
     
-    // If there's no certificate in the keychain, return nil. adError won't be set if the
-    // cert can't be found since this isn't considered an error condition.
-    if (!certificate)
-    {
-        return nil;
-    }
+    // Get the wpj certificate
+    MSID_LOG_VERBOSE(context, @"Retrieving WPJ certificate reference.");
+    status = SecIdentityCopyCertificate(identity, &certificate);
+    CHECK_KEYCHAIN_STATUS(@"Failed to read WPJ certificate.");
     
     certificateSubject = (__bridge_transfer NSString*)(SecCertificateCopySubjectSummary(certificate));
     certificateData = (__bridge_transfer NSData*)(SecCertificateCopyData(certificate));
     
-    // Get the private key
-    MSID_LOG_VERBOSE(context, @"Retrieving WPJ private key reference.");
-    
-    privateKey = [self copyPrivateKeyRefForIdentifier:kADALPrivateKeyIdentifier context:context error:&adError];
-    if (adError)
+    if (issuer)
     {
-        if (error)
-        {
-            *error = adError;
-        }
-        MSID_LOG_ERROR(context, @"Failed to retrieve WPJ private key reference. Error code %ld", (long)adError.code);
-        goto _error;
+        certificateIssuer = [[NSString alloc] initWithData:issuer encoding:NSASCIIStringEncoding];
     }
     
-    if (!identity || !certificateIssuer || !certificateSubject || !certificateData || !privateKey)
+    // Get the private key
+    MSID_LOG_VERBOSE(context, @"Retrieving WPJ private key reference.");
+    status = SecIdentityCopyPrivateKey(identity, &privateKey);
+    CHECK_KEYCHAIN_STATUS(@"Failed to read WPJ private key for identifier.");
+    
+    if (!certificate || !certificateIssuer || !certificateSubject || !certificateData || !privateKey)
     {
-        // The code above will catch missing security items, but not missing item attributes. These are caught here.
-        ADAuthenticationError* adError = [ADAuthenticationError unexpectedInternalError:@"Missing some piece of WPJ data" correlationId:context.correlationId];
         if (error)
         {
+            // The code above will catch missing security items, but not missing item attributes. These are caught here.
+            ADAuthenticationError* adError =
+            [ADAuthenticationError unexpectedInternalError:@"Missing some piece of WPJ data"
+                                             correlationId:context.correlationId];
+            
             *error = adError;
         }
+        
         goto _error;
```

Comment:
the question I had is where else this goto is being used other than this one place?

Reply:
This goto is used in the macro as discussed offline.